ssm@Severga/files/ssm@Severga/applet.js

```diff
@@ -0,0 +1,452 @@
+// Simplified System Monitor Applet
+
+const AppletUUID = "ssm@Severga";
+
+const Applet = imports.ui.applet;
+const St = imports.gi.St;
+const Clutter = imports.gi.Clutter;
+const Lang = imports.lang;
+const Mainloop = imports.mainloop;
+const Settings = imports.ui.settings;
+const GLib = imports.gi.GLib;
+const Util = imports.misc.util;
+
+let GTop, failed = false;
+try {
+	GTop = imports.gi.GTop;
+}
+catch (e) {
+	global.logError("Required GTop package missing!");
+	failed = true;
+}
+
+function align(str, size) {
+	str = String(str);
+	return (str.length < size ? " ".repeat(size - str.length) + str : str);
+}
+
+
+// <SSMApplet>
+function SSMApplet(metadata, orientation, panel_height, instance_id) {
+	this._init(metadata, orientation, panel_height, instance_id);
+}
+
+SSMApplet.prototype = {
+	__proto__: Applet.Applet.prototype,
+	
+	_init: function(metadata, orientation, panel_height, instance_id) {
+		Applet.Applet.prototype._init.call(this, orientation, panel_height, instance_id);
+		
+		this.setAllowedLayout(Applet.AllowedLayout.HORIZONTAL);
+		if (failed) {
+			this.errorLabel = new St.Label();
+			this.errorLabel.set_text("Error!");
+			this.actor.add(this.errorLabel, { x_align: St.Align.MIDDLE, y_align: St.Align.MIDDLE, y_fill: false });
+			this._applet_tooltip.set_text("Required GTop package missing!");
+		} else {
+			this._applet_tooltip._tooltip.set_style('text-align: left; font-family: monospace; font-size: 9pt');
+			this._applet_context_menu.addMenuItem(new Applet.MenuItem("Clear Cache", "user-trash", () => {
+				let [success, argv] = GLib.shell_parse_argv("pkexec sh -c 'echo 1 >/proc/sys/vm/drop_caches'");
+				let flags = GLib.SpawnFlags.SEARCH_PATH | GLib.SpawnFlags.DO_NOT_REAP_CHILD;
+				let [result, pid] = GLib.spawn_async(null, argv, null, flags, null);
+			}));
+			
+			this.waitingForSensors = false;
+			this.sensorsFailed = false;
+			this.currentTemp = -283;
+			this._netBlackList = [];
+			
+			//this.sensorsPath = "/usr/bin/sensors";
```

Comment:
With this commented out do you really need to have lm-sensors as a requirement?

Reply:
I'd been using these lines before I add the settings support. Such commented out lines in the code may be deleted without any harm, but I left them just for myself to remember the names of my variables and what they contain. This property (sensorsPath)'s value is now taken from the applet settings. Maybe I was wrong including this setting. But I wasn't sure my code would be able to find "sensors" executable without user's help. :) My knowledge of Mint's libraries is very bad.

Anyway, you are right. The applet can work without "lm-sensors" at all. And it will do it without errors. Just there will be no color changes according to the CPU's temperature. So I should find other term instead of "requirement".